parser.go

```diff
@@ -358,10 +358,11 @@ func (p *parser) parseEntryPoint() error {
 		return fmt.Errorf("can not parse general API information: %v", err)
 	}
 
-	// Security Scopes are defined at a different level in the hierarchy as where they need to end up in the OpenAPI structure,
+	// Security scopes and tag groups are defined at a different level in the hierarchy as where they need to end up in the OpenAPI structure,
 	// so a temporary list is needed.
 	oauthScopes := make(map[string]map[string]string, 0)
-
+	tagGroups := make(map[string][]string)
+	tagGroupKeys := make([]string, 0, len(tagGroups))
```

Comment:
What is the expectation if a user has a duplicate tag name?

Reply:
```
"x-tagGroups": [
    {
      "name": "Group 1",
      "tags": [
        "Foo",
        "Bar"
      ],
      "x-traitTag": true
    },
    {
      "name": "Group 1",
      "tags": [
        "Vfoo",
        "Foo With Spaces"
      ],
      "x-traitTag": true
    }
  ]
```

I believe this would be an incorrect input, is that right? Just wanted to make sure we could fail gracefully.

Reply:
yeah good question, this would be incorrect _output_. The input is defined in the examples folder in `main.go`

```
// @Tags "Group 1" "Foo" "description"
// @Tags "Group 1" "Bar" "Baz"
// @Tags "Group 2" "Vfoo" "another description"
// @Tags "Group 2" "Foo With Spaces" "some other description"
```

and the logic on ~L540 of this file checks to see if we've come across a tag group, and if we have, just append the tag to that tag group, and if we haven't, add a new key to the tag group map/key to the array